nifi-extension-bundles/nifi-github-bundle/nifi-github-extensions/src/main/java/org/apache/nifi/github/GitHubRepositoryClient.java

```diff
@@ -276,8 +299,13 @@ public List<GitCommit> getCommits(final String path, final String branch) throws
                         .toList();
 
                 final List<GitCommit> commits = new ArrayList<>();
+                int i = 0;
                 for (final GHCommit ghCommit : ghCommits) {
+                    if (i >= MAX_COMMITS_TO_RETRIEVE) {
+                        break;
+                    }
```

Comment:
Limiting this after retrieval does not seem like the best approach. The Page Size appears to limit the current number to 50. Would it be sufficient to switch that to 10?

Reply:
The page size does not limit how many commits we retrieve, only how many commits we have per page but the listing would go over all of the pages regardless. This is something in the implementation of the client itself. The only way to limit the amount of commits retrieved during listing would be by providing an "afterDate" value in the query builder.

Reply:
Made a small change to use the client in order to only retrieve the first page of commits and only define a page size.